compliance_checker/cf/cf.py

```diff
@@ -2360,66 +2360,78 @@ def check_packed_data(self, ds):
+                reasoning.append("Attributes add_offset and scale_factor have different data type.")
+            elif type(scale_factor) != var.dtype:
+                # Check both attributes are type float or double
+                if not type(scale_factor) in [np.float, np.float16, np.float32, np.float64, np.float128]:
+                    valid = False
+                    reasoning.append("Attributes add_offset and scale_factor are not of type float or double.")
+                else:
+                    # Check variable type is byte, short or int
+                    if not var.dtype in [np.int, np.int8, np.int16, np.int32, np.int64]:
                         valid = False
-                        reasoning.append("'add_offset' and 'scale_factor' are not both of type float or int or the data variable is not of type byte, short, or int.")
-                        result = Result(BaseCheck.MEDIUM,                            \
-                                valid,                                       \
-                                ('var', name, 'packed_data'), \
-                                reasoning)
-                        ret_val.append(result) 
-                        reasoning = []
+                        reasoning.append("Variable is not of type byte, short, or int.")
 
+            result = Result(BaseCheck.MEDIUM, valid, ('var', name, 'packed_data'), reasoning)
+            ret_val.append(result)
+            reasoning = []
 
+            valid = True
+            # test further with  _FillValue , valid_min , valid_max , valid_range
+            if hasattr(var, "_FillValue"):
+                if var._FillValue.dtype !=  var.dtype:
+                    valid = False
+                    reasoning.append("Type of _FillValue attribute (%s) does not match variable type (%s)" %\
+                                     (var._FillValue.dtype, var.dtype))
+            if hasattr(var, "valid_min"):
+                if var.valid_min.dtype != var.dtype:
+                    valid = False
+                    reasoning.append("Type of valid_min attribute (%s) does not match variable type (%s)" %\
+                                     (var.valid_min.dtype, var.dtype))
+            if hasattr(var, "valid_max"):
+                if var.valid_max.dtype != var.dtype:
+                    valid = False
+                    reasoning.append("Type of valid_max attribute (%s) does not match variable type (%s)" %\
+                                     (var.valid_max.dtype, var.dtype))
+            if hasattr(var, "valid_range"):
+                if var.valid_range.dtype != var.dtype:
+                    valid = False
+                    reasoning.append("Type of valid_range attribute (%s) does not match variable type (%s)" %\
+                                     (var.valid_range.dtype, var.dtype))
+
+            if not valid:
+                result = Result(BaseCheck.MEDIUM, False, ('var', name, 'packed_data'), reasoning)
```

Comment:
It would be good for the results from this whole section always be added, instead of only added on failures.  Basically, if you check anything, a `Result` should be produced.  I know we're not entirely consistent with that throughout the codebase but should be.